packages/playwright-test/src/reporters/html.ts

```diff
@@ -142,6 +145,25 @@ class HtmlReporter implements Reporter {
 
   onBegin(config: FullConfig, suite: Suite) {
     this.config = config as FullConfigInternal;
+    const { outputFolder, open } = this._resolveOptions();
+    this._outputFolder = outputFolder;
+    this._open = open;
+    const reportedWarnings = new Set<string>();
+    for (const project of config.projects) {
+      if (outputFolder.startsWith(project.outputDir) || project.outputDir.startsWith(outputFolder)) {
```

Comment:
can someone explain the condition there? from what I understand the default test result is `test-results` and if I configure the html reporter with ` outputFolder: 'test-results/html-report'` then I should not see this error: 

![image](https://user-images.githubusercontent.com/5362281/183532550-9ac311ee-6a7f-4be5-b069-67837b5d2ac9.png)

because in the `onEnd` method [here](https://github.com/microsoft/playwright/pull/14964/files#diff-16ac82cfcd6cc32aee0a06c1f9f86c2c26d3cfc20857225da0fd3395e324cc06R187) we are removing the html folder and not the parent `test-results` folder